docs/Operators.md

```diff
@@ -7630,39 +7826,79 @@ This version of the operator has been available since version 16 of the default
+<dt><tt>nearest_mode</tt> : string (default is round_to_even)</dt>
+<dd>This attribute specifies the rounding mode used in nearest interpolation and must be one of the strings 'round_prefer_floor' (round .5 down), 'round_prefer_ceil' (round .5 up), 'round_to_even' (round .5 to closest even integer). If the rounding mode is not nearest then this attribute is ignored, though it must still be one of the indicated strings.</dd>
 </dl>
 
+
+
 #### Inputs
 
 <dl>
-<dt><tt>X</tt> (differentiable) : T1</dt>
-<dd>4-D tensor of shape (N, C, H, W), where N is the batch size, C is the numbers of channels, H and W are the height and width of the input data.</dd>
-<dt><tt>grid</tt> (non-differentiable) : T1</dt>
-<dd>Input offset, 4-D tensor of shape (N, H_out, W_out, 2), where H_out and W_out are the height and width of grid and output, Grid specifies the sampling pixel locations normalized by the input spatial dimensions. Therefore, it should have most values in the range of [-1, 1]. If grid has values outside the range of [-1, 1], the corresponding outputs will be handled as defined by padding_mode.</dd>
+<dt><tt>input</tt> (differentiable) : T1</dt>
+<dd>4-D tensor of shape (N, C, H, W), where N is the batch size, C is the numbers of channels, H and W are the height and width of the input data. This is the tensor that GridSample does lookups in to retrieve data that is used to compute the output. It is not allowed for N, C, H or W to be zero.</dd>
+
+<dt><tt>grid</tt> (non-differentiable) : T2</dt>
+<dd>4-D tensor of shape (N, H_out, W_out, 2), where H_out and W_out are the height and width of grid and output. This is the tensor that GridSample uses to determine where to do lookups in input. This parameter is allowed to be integer, but this would typically indicate a mistake as this can only access the input data at its corners. It is not allowed for N, H_out or W_out to be zero.</dd>
 </dl>
 
 #### Outputs
 
 <dl>
-<dt><tt>Y</tt> (differentiable) : T2</dt>
+<dt><tt>output</tt> (differentiable) : T1</dt>
 <dd>4-D tensor of shape (N, C, H_out, W_out).</dd>
 </dl>
 
 #### Type Constraints
 
 <dl>
-<dt><tt>T1</tt> : tensor(uint8), tensor(uint16), tensor(uint32), tensor(uint64), tensor(int8), tensor(int16), tensor(int32), tensor(int64), tensor(float16), tensor(float), tensor(double), tensor(string), tensor(bool), tensor(complex64), tensor(complex128)</dt>
-<dd>Constrain input types to all tensor types.</dd>
+<dt><tt>T1</tt> : tensor(float16), tensor(float), tensor(double), tensor(complex64), tensor(complex128)</dt>
+<dd>Constrain input and output types to float and complex tensors.</dd>
 <dt><tt>T2</tt> : tensor(float16), tensor(float), tensor(double)</dt>
-<dd>Constrain output types to float tensors.</dd>
+<dd>Constrain grid operand type to float tensors.</dd>
 </dl>
 
+#### Unresolved
+
+**Some concerns from the issue reported on GridSample that are not resolved here:
+
+1. We need dimensions higher than 4D for GridSample. Can the spec run
```

Comment:
Thanks for the detailed and clear issue contributions. That's a very good idea for 5D support. But if we add the 5d spec description, users who are not familiar with this op would be confused if they export their model with it in 5D and failed. So it's not a good time to add it at present. And welcome to contribute the 5D kernel implementation of GridSample.

Reply:
I kept it to 4D in this pull request.